lua/neo-tree/ui/renderer.lua

```diff
@@ -522,8 +522,10 @@ M.focus_node = function(state, id, do_not_focus_window, relative_movement, botto
     end
     local success, err = pcall(vim.api.nvim_win_set_cursor, state.winid, { linenr, col })
 
-    -- now ensure that the window is scrolled correctly
     if success then
+      -- forget about cursor position as it is overwritten
+      M.position.clear(state)
```

Comment:
Couldn't we have just set 
```suggestion
      M.position.is.restorable = false
```
 at this location and accomplish the same thing without adding the clear method? Are we duplicating functionality by have both?

Reply:
No, we need to make `topline` or `lnum` falsy to skip the first line in save function.
Due to how the code is structured, the oldest save is most respected now. As I've written in the comment, when cursor position is overwritten, we should forget about this so that a later call to save can actually save its value. Otherwise, if the position is still saved, then the next call will not overwrite since oldest is greatest.

Additionally, I thought these `state.position.*` variables as private properties, and didn't like the idea to modify them from outside the scope.
That's why I made a public facing method to change these internal properties. Just like getters and setters in other languages.

Reply:
I personally prefer the clear method to the `is.restorable` property, but it seems like we should not have both. Can we get rid of the `is.restorable` or is it handling another situation?

Reply:
As the comment says.

https://github.com/nvim-neo-tree/neo-tree.nvim/blob/db178f4a49c19f8e4ed5a01dafa9d79e76f0081e/lua/neo-tree/ui/renderer.lua#L657-L659

This variable is more to decide whether we should restore the cursor position in the next iteration.
What `.clear()` does is more of a `is.savable + is.setable` (`save()` AND `set()` will do the thing next time), which stores if the next call to save should overwrite the previous saved `topline / lnum`.

But I don't like the name _savable_, like looking at the variable name, I don't quite get what the variable means at glance, or I have to read thru all the discussions that's held now to get the context.
Whereas `clear()` (to me) sounds like it forgets about the current cursor position and that's what it's literally doing.

Reply:
I still don't see how setting `is.restorable = false` has a different outcome from using `clear()` in that situation and removing all references to `is.restorable`. If there is a subtle difference, then the system is too complicated.

Reply:
Do you mean to depreciate `is.restorable` and use clear on every place? Then that's no problem, clear is a superset of that.

Reply:
Yes, but I don't think there is no need to deprecate as this is all completely internal and shouldn't be used by end users. I searched and no external source I know of use it either.

Reply:
Ok, I'll delete all occurrences of is.restorable and replace it with clear. 

Plz wait for a couple of days due to real life issues.

Reply:
Done @cseickel ;)